src/Features/Core/Portable/BraceCompletion/AbstractBraceCompletionService.cs

```diff
@@ -0,0 +1,178 @@
+
+        ///<inheritdoc cref="IBraceCompletionService.AllowOverTypeAsync(BraceCompletionContext, CancellationToken)"/>
+        public abstract Task<bool> AllowOverTypeAsync(BraceCompletionContext braceCompletionContext, CancellationToken cancellationToken);
+
+        ///<inheritdoc cref="IBraceCompletionService.GetBraceCompletionAsync(BraceCompletionContext, CancellationToken)"/>
+        public async Task<BraceCompletionResult?> GetBraceCompletionAsync(BraceCompletionContext braceCompletionContext, CancellationToken cancellationToken)
+        {
+            var closingPoint = braceCompletionContext.ClosingPoint;
+            if (closingPoint < 1)
+            {
+                return null;
+            }
+
+            var openingPoint = braceCompletionContext.OpeningPoint;
+            var document = braceCompletionContext.Document;
+
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (sourceText[openingPoint] != OpeningBrace)
+            {
+                return null;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(openingPoint, findInsideTrivia: true);
+            var validOpeningPoint = await IsValidOpenBraceTokenAtPositionAsync(token, openingPoint, document, cancellationToken).ConfigureAwait(false);
+            if (!validOpeningPoint)
+            {
+                return null;
+            }
+
+            var braceTextEdit = new TextChange(TextSpan.FromBounds(closingPoint, closingPoint), ClosingBrace.ToString());
+            // The caret location should be right before where the closing brace was inserted.
+            return new BraceCompletionResult(ImmutableArray.Create(braceTextEdit), caretLocation: openingPoint + 1);
+        }
+
+        ///<inheritdoc cref="IBraceCompletionService.GetTextChangesAfterCompletionAsync(BraceCompletionContext, CancellationToken)"/>
+        public virtual Task<BraceCompletionResult?> GetTextChangesAfterCompletionAsync(BraceCompletionContext braceCompletionContext, CancellationToken cancellationToken)
+            => SpecializedTasks.Default<BraceCompletionResult?>();
+
+        ///<inheritdoc cref="IBraceCompletionService.GetTextChangeAfterReturnAsync(BraceCompletionContext, bool, CancellationToken)"/>
+        public virtual Task<BraceCompletionResult?> GetTextChangeAfterReturnAsync(BraceCompletionContext braceCompletionContext, bool supportsVirtualSpace, CancellationToken cancellationToken)
+            => SpecializedTasks.Default<BraceCompletionResult?>();
+
+        ///<inheritdoc cref="IBraceCompletionService.IsValidForBraceCompletionAsync(char, int, Document, CancellationToken)"/>
+        public virtual async Task<bool> IsValidForBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken)
```

Comment:
```suggestion
        /// <inheritdoc cref="IBraceCompletionService.GetTextChangesAfterCompletionAsync(BraceCompletionContext, CancellationToken)"/>
        public virtual Task<BraceCompletionResult?> GetTextChangesAfterCompletionAsync(BraceCompletionContext braceCompletionContext, CancellationToken cancellationToken)
            => SpecializedTasks.Default<BraceCompletionResult?>();

        /// <inheritdoc cref="IBraceCompletionService.GetTextChangeAfterReturnAsync(BraceCompletionContext, bool, CancellationToken)"/>
        public virtual Task<BraceCompletionResult?> GetTextChangeAfterReturnAsync(BraceCompletionContext braceCompletionContext, bool supportsVirtualSpace, CancellationToken cancellationToken)
            => SpecializedTasks.Default<BraceCompletionResult?>();

        /// <inheritdoc cref="IBraceCompletionService.IsValidForBraceCompletionAsync(char, int, Document, CancellationToken)"/>
        public virtual async Task<bool> IsValidForBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken)
```

Reply:
fwiw, i don't think you need these inherit-docs.  if these are interface impls, don't we get this for free?